### PR TITLE
Export support for panoptic segmentation model

### PR DESCRIPTION
**Pull request by daigo0927**
## Motivation

Support panoptic segmentation model (especially PanopticFPN by now) from the MMDetection codebase. 

Originally I created a PR at the MMDetection repository: https://github.com/open-mmlab/mmdetection/pull/8169 and migrated to MMDeploy.

## Modification

- Add some `@FUNCTION_REWRITER.register_rewriter` for supporting PanopticFPN export.
- Add panoptic segmentation configs
- Append `semseg` to the `detector_forward` outputs.

## BC-breaking (Optional)

I have not faced any BC-breaking errors but I'm not sure that it is safe to append `semseg` to output_names at `mmdeploy/codebase/mmdet/models/detectors/base.py:__forward_impl`. I would appreciate it if you give some advice or instruction.

## Use cases (Optional)

We can export PanopticFPN of MMDetection. I have checked my modification to work via docker environment (almost the same as [docker/CPU/Dockerfile](https://github.com/open-mmlab/mmdeploy/blob/master/docker/CPU/Dockerfile)).

See details at https://github.com/daigo0927/blog/tree/master/mmdeploy-panopticfpn-onnx

## Checklist

1. Pre-commit or other linting tools are used to fix the potential lint issues.
2. The modification is covered by complete unit tests. If not, please add more unit tests to ensure the correctness.
3. If the modification has a dependency on downstream projects of a newer version, this PR should be tested with all supported versions of downstream projects.
4. The documentation has been modified accordingly, like docstring or example tutorials.

### CODE REVIEW

**Comment by RunningLeon** [comment]
@daigo0927 Hi, thanks for your contribution. We'll review this PR soon.

File: mmdeploy/codebase/mmdet/models/detectors/panoptic_two_stage_segmentor.py [hunk 1]
```diff
@@ -0,0 +1,41 @@
+# Copyright (c) OpenMMLab. All rights reserved.
+from mmdeploy.core import FUNCTION_REWRITER
+
+
+@FUNCTION_REWRITER.register_rewriter(
+    'mmdet.models.detectors.panoptic_two_stage_segmentor'
+    '.TwoStagePanopticSegmentor.simple_test')
+def two_stage_panoptic_segmentor__simple_test(ctx,
```
Comment (RunningLeon) [inline mmdeploy/codebase/mmdet/models/detectors/panoptic_two_stage_segmentor.py:8]: Could add unit test for this rewriting: https://github.com/open-mmlab/mmdeploy/blob/master/tests/test_codebase/test_mmdet/test_mmdet_models.py

File: mmdeploy/codebase/mmdet/models/detectors/base.py [hunk 2]
```diff
@@ -6,7 +6,9 @@
 
 
 @mark(
-    'detector_forward', inputs=['input'], outputs=['dets', 'labels', 'masks'])
+    'detector_forward',
+    inputs=['input'],
+    outputs=['dets', 'labels', 'masks', 'semseg'])
```
Comment (RunningLeon) [inline mmdeploy/codebase/mmdet/models/detectors/base.py:11]: To make the exported model run inference and visualize, we may need to update the `End2EndModel`. Here we suppose three output, may expand to four outputs?

https://github.com/open-mmlab/mmdeploy/blob/fa034e042bb435ad02eedb0961e4672256b42c64/mmdeploy/codebase/mmdet/deploy/object_detection_model.py#L201

**Comment by RunningLeon** [comment]
@daigo0927 Hi, hope you could update according to the left comments. Any discussion is welcomed.

**Comment by daigo0927** [comment]
Hi @RunningLeon , thank you for the kind message. I am a little busy on my job but willing to update this PR. Thanks!

File: mmdeploy/codebase/mmdet/models/detectors/base.py (hunk 2 above)
Comment (daigo0927) [reply mmdeploy/codebase/mmdet/models/detectors/base.py:11]: OK. I'll add an additional operation for treating 4th element of output (semantic segmentation) for panoptic segmentation.

File: mmdeploy/codebase/mmdet/models/detectors/panoptic_two_stage_segmentor.py (hunk 1 above)
Comment (daigo0927) [reply mmdeploy/codebase/mmdet/models/detectors/panoptic_two_stage_segmentor.py:8]: Thanks. Should I implement `test_two_stage_panoptic_segmentor` (for above)? or `test_base_semantic_head` (for another re-written `seg_heads.base_semantic_head.BaseSemanticHead`) or both?

**Comment by RunningLeon** [comment]
@daigo0927 Hi, sorry to bother you, but do you have time for this PR?

**Comment by Kenneth-X** [comment]
does it support mask2former or just panoptic FPN ?

**Comment by daigo0927** [comment]
@RunningLeon Thanks for your reminder. I'm afraid I don't have much time to update this PR. I would close this PR 😢 .

**Comment by Likhith-Sugganahalli** [comment]
@RunningLeon Hi, i would like to complete this pull and get panoptic deployment working with mmdeploy. i checkout out the repo @daigo0927 made, and i see he was able to mostly complete the changes, but i am unable to find any documentation on supporting new models other then a very brief explanation of the rewriter for unsupported functions, do you have something a little more detailed on the model conversion. A document going over the steps involved would be great

**Pull request closed without merging**